### DISCUSSION TITLE
Allow some field settings to be overridden from the field layout designer

### DISCUSSION THREAD

**Discussion by jlawrence-yellostudio** [craftcms/cms#8688] [Ideas] [answered]
If I create a field for example "Related Entry" with a limit of 1, I can only use this in other field layouts where only one related entry is also desired. Resulting in me having to create multiple fields for essentially the same purpose, with slight variations in the settings.

If the Limit could be set on the field layout, that would be great; I don't expect this to be easy as "limit" isn't a global setting same as the "required" flag is. 

Maybe you could add an extra column in the fieldlayoutfields table which could accommodate some non global settings and grab the limit from there? 

What do you think @brandonkelly ?

**Comment by brandonkelly** [comment]
I’m not sure I’m following. The “Branch Limit” setting is currently only available on Categories fields, not Users fields. Is “Related User” a Categories field? Or did you mean the “Limit” setting?

**Comment by jlawrence-yellostudio** [comment]
@brandonkelly Apologies I meant limit, not branch limit (i've edited for clarity).

My real life example was actually to do with entries, I have to create an additional field pointing at exactly the same entry source for the same purpose just because the limit is global to that field.

**Comment by brandonkelly** [comment]
Gotcha. It’s similar to #806 (ability to rename fields from the field layout designer) and #841 (ability to override the field instructions from the field layout designer).

We have plans to explore all of this (including overriding field type-specific settings) from the field layout designer as part of a larger content refactoring project we’re planning for Craft 4 (#1524).

Good to have specific issues we can reference as we’re working on the changes, though, so I will keep this open and add it to the 4.0 milestone.

**Comment by jlawrence-yellostudio** [comment]
thanks @brandonkelly

**Comment by brandonkelly** [comment, marked as answer]
We have decided **not** to implement this feature, as it would end up being a little overly complicated. Some field types have different `content` column types depending on their settings, and it would be annoying for field types to need to somehow identify which settings are safe to be overridden, versus not.

**Update:** I reconsidered for Craft 5 since fields no longer have `content` columns, however it’s not clear how we’d tackle it from the UI.

For field instance label overrides, if you leave the input blank, the global field name will be used, and there is an explicit “Hide” checkbox you need to tick if you want the label to be blank.

If custom fields were to start including per-instance setting overrides, they would need to render them separately from the main settings, so each setting could have some similar affordance that determines whether the override value should be respected. A dropdown setting would need to start including a “Use default value” option; a checkbox list would need to be conditional based on some parent “Override [Setting Name]” toggle, etc. It would get messy.

**Reply by proimage** [reply]
Would it make sense to limit what could be overridden per-instance to just being able to **alter the initial default selection or state of a custom field instance, and not what the final value can or cannot be**?

In other words, the options and values and sources and whatnot cannot change from instance to instance—the stored value of every instance of a specific field would always be valid if it were, say, magically transposed with the value of some _other_ instance of that same field. The only things that can be customizable per-instance are what the initial defaults are set to (or other aspects which don't affect which value(s) can be chosen):
- Which dropdown or radio option is selected
- Which checkboxes are checked
- Is the lightswitch on or off by default
- What the placeholder text is
- What the "Select Assets" button text is

I'm not sure if that more-narrow scope helps resolve the UI issues you mention, but... baby steps. ;)

**Comment by jlawrence-yellostudio** [comment]
Hi @brandonkelly 
Thanks for following up on this. Still to this day my biggest gripe is things like selection limits and sources, both of which wouldn't have any different content types as they would both be for relation fields. I have to create two fields which are exactly the same to have say limit 1 vs limit 5, or to be able to restrict the element selector to a different source.

A good example of this is a generic 'gallery' field, we always end up having to have several fields basically doing the same thing just to configure those values, which also makes the templating components more complex by specifying field handles for them rather than drop in partials.

**Comment by foxtrotcz** [comment]
Hi, ability to override field values/settings per instance would be useful.
For example changing static default values in each instance of Table field.
I described it here: https://github.com/craftcms/cms/discussions/15312
Thanks.

**Comment by ryanleichty** [comment]
Coming back to this one as I still wish we could override the default option.

What we currently do is duplicate the field—in this case `buttonVariant`—and then rename the handle with the override value in the handle `buttonVariantText`. Small thing, but it would be nice to only have one field. Not a big deal either way if it's technically difficult or not worth the effort.

<img width="3680" height="2384" alt="CleanShot 2025-07-16 at 10 28 52@2x" src="https://github.com/user-attachments/assets/0155c89a-566e-42e1-829a-2a7664b8e6db" />

**Comment by adrienne** [comment]
Cosigning this; the one that bites me a lot is the number of allowed items in a relationship field or Matrix. I would have to make vastly fewer fields on a current project if I could take a field that normally allows from 0-3 relations and say "in this case only, it can have 0-5".
